lib/rouge/lexers/liquid.rb

```diff
@@ -88,6 +88,38 @@ class Liquid < RegexLexer
           push :variable_tag_markup
         end
 
+        # iteration
+        rule %r/
+          (for)(\s+)
+          ([\w-]+)(\s+)
+          (in)(\s+)
+          (
+            (?:
+              (?:"[^"]*"|'[^']*') |
+              (?:
+                [^\s,\|'"] |
+                (?:"[^"]*"|'[^']*')
+              )+
+            )+
+          )
+          (\s)*(reversed)?
+        /x do |m|
+          groups Name::Tag, Text::Whitespace, Name::Variable, Text::Whitespace,
+                 Keyword::Reserved, Text::Whitespace
+
+          token_class = case m[7]
+                        when %r/'[^']*'/ then Str::Single
+                        when %r/"[^"]*"/ then Str::Double
+                        else
+                          Name::Variable
+                        end
+          token token_class, m[7]
+
+          token Text::Whitespace, m[8]
+          token Name::Attribute,  m[9]
+          push :tag_markup
+        end
+
```

Comment:
This is impressive but is it necessary? We already have a rule that matches an arbitrary `end<tag>` tag. Why not just get rid of the existing `end(case|unless…)` regex?

Reply:
By "this", are you referring to a particular line or `rule` or the entire PR? Sorry, I'm not able to place a finger exactly.
The focus of this PR is not the closing `end(.*)`.

It's on highlighting the "parameter" of the `for` tag, the keyword `in` and the collection reference.

Reply:
Note: The regex is taken directly from Liquid's source code.

Reply:
The entire rule itself is impressive in its scope and coverage. I'm just a little wary about rules that make the lexer more of a parser. I'll have a bit of a play when I'm back at home.

Reply:
Sorry, took longer for me to get around to this than I originally thought. Looks good to me.